### PR TITLE
fix(components/AppHeaderBar): Use Action instead of NavItem + Tooltip…

### PR DESCRIPTION
**Pull request by ncomont**
fix(components/AppHeaderBar): Use Action instead of NavItem + Tooltip placement management in Action and Typeahead

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
AppHeaderBar uses NavItem for header buttons
Tooltips position is "top" by default and it is not possible to change it

**What is the new behavior?**
AppHeader uses Actions for header buttons
AppHeader Actions accepts "tooltipPlacement"
Typeahead accepts "tooltipPlacement"

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://330.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://330.talend.surge.sh)

File: packages/components/src/AppHeaderBar/AppHeaderBar.component.js [hunk 1]
```diff
@@ -11,18 +11,27 @@ import {
...
-export function renderNavItem(props, index) {
-	const { icon, ...rest } = props;
+export function renderNavItem(props) {
+	const { onClick, icon, name, tooltipPlacement, ...rest } = props;
+
 	return (
-		<NavItem key={index} {...rest}>
-			<Icon name={icon} />
-		</NavItem>
+		<Action
```
Comment (frassinier) [inline packages/components/src/AppHeaderBar/AppHeaderBar.component.js]: Be careful, you need a `<li />` here.

File: packages/components/src/AppHeaderBar/AppHeaderBar.component.js [hunk 2]
```diff
@@ -11,18 +11,27 @@ import {
...
+			onClick={onClick}
+			label={name}
+			hideLabel
+			icon={icon}
+			className={theme['only-icon-cls']}
```
Comment (frassinier) [inline packages/components/src/AppHeaderBar/AppHeaderBar.component.js]: Where does this className come from?

File: packages/components/src/AppHeaderBar/AppHeaderBar.component.js (hunk 1 above)
Comment (frassinier) [inline packages/components/src/AppHeaderBar/AppHeaderBar.component.js]: capture du 2017-03-29 10-35-06

**Comment by frassinier** [comment]
Could you resolve conflicts on lint output files in order to view issues please?

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://330.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://330.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://330.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://330.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://330.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://330.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://330.talend.surge.sh)

File: output/theme.sasslint.txt [hunk 3]
```diff
@@ -23,10 +23,13 @@ src/theme/_forms.scss
   311:9   error    Qualifying elements are not allowed for class selectors  no-qualifying-elements
   447:8   warning  Vendor prefixes should not be used                       no-vendor-prefixes
 
+src/theme/_navbar.scss
+  53:13  warning  Selectors must be placed on new lines  single-line-per-selector
```
Comment (frassinier) [inline output/theme.sasslint.txt]: Please don't introduce new issue

Comment (frassinier) [inline output/theme.sasslint.txt]: https://github.com/Talend/ui/pull/330/files#diff-661265a9d8cca93a9cff5ab894e4822dR53

File: packages/components/src/AppHeaderBar/AppHeaderBar.component.js [hunk 4]
```diff
@@ -7,28 +7,38 @@ import {
...
+import Action from '../Actions/Action';
 import Typeahead from '../Typeahead';
 import theme from './AppHeaderBar.scss';
 
 const NAV_ITEM = 'navItem';
 const DROPDOWN = 'dropdown';
 
-export function renderNavItem(props, index) {
-	const { icon, ...rest } = props;
+function NavListItem(props) {
```
Comment (frassinier) [inline packages/components/src/AppHeaderBar/AppHeaderBar.component.js]: Don't miss its PropTypes declaration

File: packages/theme/src/theme/_navbar.scss [hunk 5]
```diff
@@ -49,10 +49,15 @@
...
 	&-nav {
-		> li > a {
+
+		> li > a, > li > button {
 			display: flex;
 			align-items: center;
 			height: $navbar-height;
 		}
+
+		li > button > i {
```
Comment (frassinier) [inline packages/theme/src/theme/_navbar.scss]: this is not used anymore

File: packages/components/src/AppHeaderBar/AppHeaderBar.component.js (hunk 4 above)
Comment (ncomont) [inline packages/components/src/AppHeaderBar/AppHeaderBar.component.js]: :ok_hand:

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://330.talend.surge.sh)

File: packages/components/src/HeaderBar/__snapshots__/HeaderBar.snapshot.test.js.snap [hunk 6]
```diff
@@ -71,6 +71,7 @@ exports[`Storyshots HeaderBar default 1`] = `
               aria-describedby={42}
               className="btn btn-link"
               disabled={false}
+              id="header-search"
```
Comment (mtikhinskaya) [inline packages/components/src/HeaderBar/__snapshots__/HeaderBar.snapshot.test.js.snap]: Why did you change tests for components you don't touch?

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://330.talend.surge.sh)

File: packages/components/src/Typeahead/Typeahead.component.js [hunk 7]
```diff
@@ -28,6 +28,8 @@ function Typeahead({ onToggle, icon, position, ...rest }) {
 				icon={icon.name}
 				bsStyle={icon.bsStyle}
 				className={theme['only-icon-cls']}
+				tooltipPlacement={icon.tooltipPlacement}
+				{...rest}
```
Comment (jmfrancois) [inline packages/components/src/Typeahead/Typeahead.component.js]: that's why :)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://330.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://330.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://330.talend.surge.sh)
